### DISCUSSION TITLE
How do you perform ctrl+tab and ctrl+shift+tab? is there a way to do alt+shift+tab with the swapper?

### DISCUSSION THREAD

**Discussion by plo009** [urob/zmk-config#62] [Q&A]
Hi there, was wondering how to you perform these actions in the browser with your keymap?

**Comment by urob** [comment]
I use the HRMs on the right hand and the tab combo on the left for that. They are all in the home-position so I find it very comfortable. 

Using the `key-labels` notation from the helpers, CTRL + TAB is:
`RM1` + `LM3` + `LM4`. CTRL + SHIFT + TAB adds `RM2` to the mix. 

Like with any shortcut, the first time you are doing it takes a second to think about. But after that it comes very intuitive. 

That said, if you want to further simplify you could also define something like this:
```c
#define TAB_RT LC(TAB)
#define TAB_LT LC(LS(TAB))
```
And add them in a convenient location (eg upper left keys on the Nav layer). Using swapper here would be a bit overkill as there is no need to hold anything down in-between tab changes.

**Reply by plo009** [reply]
Hi, thanks for taking the time to explain this to me, I'm still experimenting with with the keymap.
Forgive me if I'm asking some noob questions :), so i swapped the smart_num to the left hand side as I'm used to have space and enter on the right half.

So i personally like using left hand only for some simple shortcuts freeing up the right hand to use the mouse while i'm browsing
So here is something that compiles but doesn't work at all probably  due to my lack of understanding of ZMK.

`
ZMK_MOD_MORPH(smart_tab,
    bindings = <SMART_NUM>, <&kp TAB>;
    mods = <(MOD_LALT|MOD_LCTL)>;
    keep-mods = <(MOD_LALT|MOD_LCTL)>;
)
`

**Reply by urob** [reply]
What are you trying to achieve here? If you want one-handed tab switching, I'd just do something along the lines I outlined above and add `LC(TAB)` and `LC(LS(TAB))` onto the nav-layer (e.g., in the upper left corner).
